tests/sc_rotate_key_test.go

```diff
@@ -0,0 +1,270 @@
+package tests_test
+
+import (
+	"testing"
+
+	vgrand "code.vegaprotocol.io/shared/libs/rand"
+	"code.vegaprotocol.io/vegawallet/cmd/flags"
+	"code.vegaprotocol.io/vegawallet/wallet"
+	"github.com/stretchr/testify/require"
+)
+
+func TestRotateKeySucceeds(t *testing.T) {
+	// given
+	home, cleanUpFn := NewTempDir(t)
+	defer cleanUpFn(t)
+
+	_, passphraseFilePath := NewPassphraseFile(t, home)
+
+	walletName := vgrand.RandomStr(5)
+
+	cmd := []string{
+		"--home", home,
+		"--output", "json",
+		"--wallet", walletName,
+		"--passphrase-file", passphraseFilePath,
+	}
+
+	// when
+	generateKeyResp1, err := KeyGenerate(t, append(cmd,
+		"--meta", "name:key-1,role:validation",
+	))
+
+	// then
+	require.NoError(t, err)
+	AssertGenerateKey(t, generateKeyResp1).
+		WithWalletCreation().
+		WithName(walletName).
+		WithMeta(map[string]string{"name": "key-1", "role": "validation"}).
+		LocatedUnder(home)
+
+	// when
+	resp, err := KeyRotate(t, append(cmd,
+		"--public-key", generateKeyResp1.Key.KeyPair.PublicKey,
+		"--tx-height", "20",
+		"--target-height", "25",
+	))
+
+	// then
+	require.NoError(t, err)
+	AssertKeyRotate(t, resp)
```

Comment:
Is it possible to check the returned values, such as the computed signature ?

Reply:
It is in a slightly complicated way (because it has nonce so the whole string can't be asserted). But it is already checked in unit test of that function. This is actually something I wanted to discuss with you - what should go to integration tests and what should be in the handler unit test, because there is no point to write the same test twice.